[Feat] MapScreen UI

null4848 opened the pull request:
구글맵 불러오기 + UI 그리기
그리고 간단한 기능 구현

**Check List :memo:**
- [x] Pull Request 제목 : [유형] 변경 사항 요약
- [x] 설명 & 이슈 번호 작성

**설명 (필수)**

이 PR에서 어떤 변경을 했는지 상세히 설명해주세요.
관련된 Issue 번호를 포함하세요.

1. #2 

2. 변경 사항

좌측 상단에 드롭다운 배치

우측 상단에 핀 설명 박스 배치

우측 하단에 내 위치로 이동 버튼 배치
-> 하단 박스에 의해 가려짐 이슈 개선 (하단 박스 클릭시 닫히게 함으로 보이게끔 구현)
내 위치 버튼 UI 수정

처음 맵 클릭시 Navy 핀, 박스 나오게 구현
박스 버튼 클릭시 pin, box 바뀌게끔 함
Navy-> Red -> Yellow -> Blue

권한 허용 후 앱 종료 개선

SVG 파일 비트맵 변환, 색상별로 함수화 후 적용

하단 위치 정보 박스 boxes.kt에서 다룸
-> 4가지 버전 다 뜨게함


4. 스크린샷 (선택)
![Screenshot_20250216_212614](https://github.com/user-attachments/assets/12c8ac09-59a7-4928-8062-7ea90a66e910)


---

**review 참고 사항 (선택)**

추가로 알릴 내용이 있다면 여기에 작성해주세요.

1.

2.

null4848 committed:
chore: googlemap + ui

구글맵 불러오기 + UI 그리기
그리고 간단한 기능 구현

null4848 committed:
Merge remote-tracking branch 'origin/4-chore-fontsystem' into 2-feat-mapscreen-ui

null4848 committed:
refactoring: font change + agp

폰트 변경 + AGP 버전 업그레이드

moon0727 commented on app/src/main/res/drawable/markedblue.png:
Png 파일로 하신 이유가 있을까요? 가능하다면 SVG 파일로 변경해주세요!

null4848 replied:
이 질문에 일괄 답변하겠습니다. 구글 맵에서는 이미지를 Bitmap 형식으로 처리해야 해서 SVG 파일을 적용하려면 형식 변환이 필수더라구요... 이 변환 과정이 조금 복잡하고 코드가 길어져서 PNG 형식으로 진행하는 것이 더 효율적이라고 생각했습니다. 코드가 길어지더라도 (+복잡) SVG 파일로 진행할까요?

moon0727 replied:
Icon(
    imageVector = ImageVector.vectorResource(id = R.drawable.location_icon),
    contentDescription = "Location",
    tint = TextLightGray
)

이런식으로 Icon으로 구성하면 어때요? figma에서 svg로 export하고 res>drawable에 vector assets에 추가하면 사용가능합니다! 이미지 품질에 대한 걱정때문에 svg로 하는게 어떻냐고 제안드린거에요! 또 모양은 동일하니 색상만 다르게 적용하면 된다고 생각했습니다! merkers.kt 만들어서 모아두고 import해서 사용해도 괜찮구요!

null4848 replied:
저희가 사용하는 컴포즈 버전에서는 vectorresource를 지원안하더라구요!! 근데 저도 말씀해주신 것처럼 색상만 바꾸는 방식으로 svg가 맞는 것 같아서 그렇게 구현해볼게요!!

moon0727 commented on app/src/main/res/drawable/markedred.png:
Png 파일로 하신 이유가 있을까요? 가능하다면 SVG 파일로 변경해주세요!

moon0727 commented on app/src/main/res/drawable/markedyellow.png:
Png 파일로 하신 이유가 있을까요? 가능하다면 SVG 파일로 변경해주세요!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/MapScreen.kt:
```diff
@@ -1,10 +1,479 @@
 fun MapScreen() {
-    Text(text = "🗺️ Map Screen")
+    var expanded by remember { mutableStateOf(false) }
+    var selectedItem by remember { mutableStateOf("선택하세요") }
```
기본 옵션은 전체입니다!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/MapScreen.kt:
```diff
@@ -1,10 +1,479 @@
+        }
+
+        // 내 위치 버튼 → 오른쪽 하단에 배치
+        Box(
```
요거 추가하실거면 방문(혹은 찜, 혹은 기록) 하단 박스 생길 때 위로 올라오게끔 해주세요! 가려서 안보여요

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/MapScreen.kt:
```diff
@@ -1,10 +1,479 @@
+                        Spacer(modifier = Modifier.width(4.dp))
+
+                        Text(
+                            text = selectedAddress,
```
주소 한글로 변환 가능할까요?

null4848 replied:
주소 AVD 언어 설정을 한국어로 하면 변환됩니다! 기본 언어에 따라서 달라져요! 제가 AVD 초기화하고 설정 안 한 다음에 해서 이미지에는 영어로 뜨네요

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/MapScreen.kt:
```diff
@@ -1,10 +1,479 @@
+                        Spacer(modifier = Modifier.width(4.dp))
+
+                        Text(
+                            text = "남은 시간",
```
이거 임시로 3Day로 변경해주세요!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/MapScreen.kt:
```diff
@@ -1,10 +1,479 @@
+                            contentDescription = "남은 시간",
+                            modifier = Modifier.size(15.dp)
+                        )
+                        Spacer(modifier = Modifier.width(4.dp))
```
피그마 상 5dp 입니다!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/MapScreen.kt:
```diff
@@ -1,10 +1,479 @@
+                                contentColor = White
+                            )
+                        ) {
+                            Text("방문하기", style = AppTypography.labelMedium, color = White)
```
BodyMedium 입니다!

moon0727 reviewed (changes requested):
전체적으로 분리가 좀 필요해보입니다! 맵이라 어려운 부분은 있지만 간단하게 찜|기록|방문 세가지 상태에 따른 하단 박스는 밖으로 빼서 작성해도 될 것 같아요! 어렵다면 목데이터를 만들어서 테스트하면 쉬울거에요!

제가 말하는 하단 박스입니다!
<img width="290" alt="image" src="https://github.com/user-attachments/assets/ec3b51e0-d228-4539-a848-6a26817a8fae" />

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/MapScreen.kt:
```diff
@@ -1,10 +1,479 @@
+        }
+    }
+
+    // 권한 체크 후 실시간 위치 가져오기
```
권한 체크 후 앱이 꺼집니다!

null4848 committed:
feat: comment 부분 수정

권한 허용 후 앱 종료 개선
내 위치 버튼 하단 박스에 의해 가려짐 이슈 개선
-> 하단 박스 클릭시 닫히게 함으로 보이게끔 구현
SVG 파일 비트맵 변환, 색상별로 함수화 후 적용

null4848 committed:
feat: 코드 분리 + UI 수정

하단 위치 정보 박스 boxes.kt에서 다룸
-> 3가지 버전 다 뜨게함
내 위치 버튼 UI 수정

null4848 committed:
feat: boxes 수정 + 버튼 클릭시 박스, 핀 변경

boxes.kt -> history에서 수정했던거 가져옴

버튼 클릭시 핀, 박스 바뀌게끔 함
Red -> Yellow -> Blue
새로 맵 클릭시 빨간 핀, 빨간 박스 나오게 구현

코드 간결화를 위해 수정함

노트북 이슈로 push 함
-> navy 박스 추후 구현

null4848 committed:
Merge branch 'main' into 2-feat-mapscreen-ui

null4848 committed:
feat: 찜하기 박스 추가 + 버튼 클릭 구현

NavyBox 찜하기, 방문하기 버튼 추가

찜하기 버튼 함수 추가

찜하기 누르면 레드박스 방문하기 버튼 누르면 옐로박스 뜨게 구현

null4848 committed:
Merge branch 'main' into 2-feat-mapscreen-ui

null4848 committed:
feat: 기록하기 추가

기록하기 추가
historyscreen에서 navybox 부분 수정
박스 크기 수정 -> 주소가 두줄되면 밑 부분이 짤려서 안보임

null4848 committed:
chore: API 키 변경
